telemeta_mshs/middleware/keycloak_mw.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2017 Marcos Pereira <marcospereira.mpj@gmail.com>
+# Modified by Sairam Krish <haisairam@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+import logging
+from django.conf import settings
+from django.http.response import JsonResponse
+import keycloak
+from keycloak import KeycloakOpenID
+from keycloak.exceptions import KeycloakInvalidTokenError,\
+    raise_error_from_response, KeycloakGetError
+from rest_framework.exceptions import PermissionDenied, AuthenticationFailed, NotAuthenticated
+import json
+
+logger = logging.getLogger(__name__)
+
+class KeycloakMiddleware(object):
+
+    def __init__(self):
+        """
+        :param get_response:
+        """
+
+        self.config = settings.KEYCLOAK_CONFIG
+
+        # Read configurations
+        try:
+            self.server_url = self.config['KEYCLOAK_SERVER_URL']
+            self.client_id = self.config['KEYCLOAK_CLIENT_ID']
+            self.realm = self.config['KEYCLOAK_REALM']
+        except KeyError as e:
+            raise Exception("KEYCLOAK_SERVER_URL, KEYCLOAK_CLIENT_ID or KEYCLOAK_REALM not found.")
+
+        self.client_secret_key = self.config.get('KEYCLOAK_CLIENT_SECRET_KEY', None)
+        self.client_public_key = self.config.get('KEYCLOAK_CLIENT_PUBLIC_KEY', None)
+        self.default_access = self.config.get('KEYCLOAK_DEFAULT_ACCESS', "DENY")
+        self.method_validate_token = self.config.get('KEYCLOAK_METHOD_VALIDATE_TOKEN', "INTROSPECT")
+        self.keycloak_authorization_config = self.config.get('KEYCLOAK_AUTHORIZATION_CONFIG', None)
+
+        # Create Keycloak instance
+        self.keycloak = KeycloakOpenID(server_url=self.server_url,
+                                       client_id=self.client_id,
+                                       realm_name=self.realm,
+                                       client_secret_key=self.client_secret_key)
+
+        # Read policies
+        if self.keycloak_authorization_config:
+            self.keycloak.load_authorization_config(self.keycloak_authorization_config)
+
+        # Django
+        #self.get_response = get_response
```

Comment:
```suggestion
        #self.get_response = get_response
```
Pour un futur passage en Python 3